### ISSUE TITLE
gradient penalty of sigmoid instead of logits.

### ISSUE THREAD

**Issue by d-michele** [ChengBinJin/WGAN-GP-tensorflow#4]
https://github.com/ChengBinJin/WGAN-GP-tensorflow/blob/9f27dd98513e223546790865e7d71676539b49a6/src/wgan_gp.py#L94

Hi, I would ask you an explanation of why did you calculate the gradient on the sigmoid instead of logits. Thanks!

**Comment by ChengBinJin** [comment]
@d-michele 
According to the theory of WGAN and WGAN-GP, the Wasserstein distance is calculated using the output of the discriminator. And, a sigmoid function is not utilized at the end of the discriminator. Therefore, when calculating the gradient penalty from the discriminator, there is also no sigmoid function.

**Comment by opetliak** [comment]
@ChengBinJin  Hi, why do you return two outputs (sigmoid and linear) in basicDiscriminator?

**Comment by ChengBinJin** [comment]
@lenpetlyak The linear value is used by calculating the Wasserstein distance, and the sigmoid value is just for debugging.
